interpreter/cling/lib/Interpreter/InterpreterCallbacks.cpp

```diff
@@ -239,6 +241,27 @@ namespace cling {
     return false;
   }
 
+  void InterpreterCallbacks::CreateCodeCompleteConsumer(Interpreter* child) const {
+    ClingCodeCompleteConsumer* consumer = new ClingCodeCompleteConsumer(
+                m_Interpreter->getCI()->getFrontendOpts().CodeCompleteOpts,
+                llvm::outs());
+    CompilerInstance* childCI = child->getCI();
+    //codeCompletionCI will own consumer!
+    childCI->setCodeCompletionConsumer(consumer);
+    clang::Sema& childSemaRef = childCI->getSema();
+    childSemaRef.CodeCompleter = consumer;
+  }
+
+  void InterpreterCallbacks::GetCompletionResults(Interpreter* child,
+                                        std::vector<std::string>& completions) const {
+    // Which cast to use , fno-rtti
+    ClingCodeCompleteConsumer* consumer =
+                                  (ClingCodeCompleteConsumer*)(
+                                  child->getCI()->getSema().CodeCompleter);
+    if (consumer)                              
+      consumer->getCompletions(completions);
```

Comment:
This is a bit awkward - the fact that `GetCompletionResults()` needs to be called after calling the interpreter's `Complete()`. Can we
- pass the vector of strings into `Interpreter::Complete()`?
- have `Interpreter::Complete()` "register" that string with `ClingCodeCompleteConsumer` such that it adds to it in `ProcessCodeCompleteResults`?